src/edg.rs

```diff
@@ -442,43 +609,60 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
             return;
         }
         // Line 3
-        self.assignment.insert(vertex.clone(), assignment);
-        // Line 4
-        if let Some(interested) = self.interests.get(&vertex) {
-            for worker_id in interested {
-                let (new_weight, split_weight) = weight
-                    .split()
-                    .unwrap_or_else(|_| panic!("Ran out of weight on worker {}", self.id));
-                weight = new_weight;
-                self.broker.send(
-                    *worker_id,
-                    Message::ANSWER {
-                        vertex: vertex.clone(),
-                        assignment,
-                        weight: split_weight,
-                    },
-                )
+        let prev_assignment = self.assignment.insert(vertex.clone(), assignment);
+        let changed_assignment = (prev_assignment != Some(assignment));
```

Comment:
Fixes `clippy` warning
```suggestion
        let changed_assignment = prev_assignment != Some(assignment);
```